cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories.go

```diff
@@ -358,6 +395,53 @@ func (s *Server) updateRepo(ctx context.Context, repo *HelmRepository) (*corev1.
 	}, nil
 }
 
+func handleImagesPullSecret(ctx context.Context, typedClient kubernetes.Interface,
+	isUserManagedSecrets bool, repo *HelmRepository, newRepo bool) (imgPullSecret *k8scorev1.Secret, updateImgPullSecret bool, err error) {
+	if repo.customDetails == nil || repo.customDetails.ImagesPullSecret == nil {
+		if !isUserManagedSecrets {
+			managedSecretName := imagesPullSecretName(repo.name.Name)
+			secretsInterface := typedClient.CoreV1().Secrets(repo.name.Namespace)
+			if deleteErr := deleteSecret(ctx, secretsInterface, managedSecretName); deleteErr != nil {
```

Comment:
If not found, it will yield an error and we won't be able to install the repo, AFAIK